src/components/withNavigationFallback.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, forwardRef, useContext, useMemo} from 'react';
+import {NavigationContext} from '@react-navigation/core';
+import {NavigationProp} from '@react-navigation/native';
+import {ParamListBase} from '@react-navigation/routers';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+type AddListenerCallback = () => void;
+
+type RemoveListenerCallback = () => void;
+
+type NavigationContextValue = {
+    isFocused: () => boolean;
+    addListener: () => AddListenerCallback;
+    removeListener: () => RemoveListenerCallback;
+};
+
+export default function <TProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    function WithNavigationFallback(props: TProps, ref: ForwardedRef<TRef>) {
+        const context = useContext(NavigationContext);
+
+        const navigationContextValue: NavigationContextValue = useMemo(() => ({isFocused: () => true, addListener: () => () => {}, removeListener: () => () => {}}), []);
+
+        return context ? (
+            <WrappedComponent
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...props}
+                ref={ref}
+            />
+        ) : (
+            <NavigationContext.Provider value={navigationContextValue as unknown as NavigationProp<ParamListBase>}>
```

Comment:
Why we are typing `navigationContextValue` as NavigationContextValue and then changing it to `NavigationProp<ParamListBase>}`

Reply:
ok, let me check if there is a way to improve it

Reply:
Didnt have a better idea so LGTM :D maybe @blazejkustra will have